examples/inspection/plot_partial_dependence.py

```diff
@@ -360,13 +518,14 @@
 surf = ax.plot_surface(XX, YY, Z, rstride=1, cstride=1, cmap=plt.cm.BuPu, edgecolor="k")
 ax.set_xlabel(features[0])
 ax.set_ylabel(features[1])
-ax.set_zlabel("Partial dependence")
 # pretty init view
 ax.view_init(elev=22, azim=122)
-plt.colorbar(surf)
+clb = plt.colorbar(surf)
+clb.ax.set_title("Partial dependence")
 plt.suptitle(
-    "Partial dependence of house value on median\n"
-    "age and average occupancy, with Gradient Boosting"
+    "Partial dependence of number of bike rentals on\nthe temperature and humidity "
+    "gradient boosting decision tree model",
+    fontsize=16,
 )
 plt.subplots_adjust(top=0.9)
```

Comment:
Here again the layout has overlapping texts:

![image](https://user-images.githubusercontent.com/89061/203121976-dd0c7e65-c380-43a9-ace6-888fc0dba50d.png)


Let's see if using constrained layout can automatically fix that class of problems similarly.